### PR TITLE
PIMS-1823: Fix count for server-side pagination

### PR DESCRIPTION
**Pull request by LawrenceLau2020**
## 🎯 Summary

  
[PIMS-1823: ](https://citz-imb.atlassian.net/browse/PIMS-1823) - Fix count for server-side pagination

## Changes
- Used a separate query for the "totalCount" in the project and property services because the TypeORM "findAndCount" function still returns the "filtered" number of rows and not the actual count.
- Modified the FilterSearchDataGridProps to take a rowCountProp number as a property that can be passed in.
- You should be able to see the total count reflected in the properties and projects list page as well as a snackbar notification at the bottom of the screen. This number should change when applying a filter on the list being displayed.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/src/services/projects/projectsServices.ts [hunk 1]
```diff
@@ -816,6 +816,8 @@ const getProjects = async (filter: ProjectFilter) => {
     });
   }
 
+  const totalCount = await query.getCount();
+
```
Comment (LawrenceLau2020) [inline express-api/src/services/projects/projectsServices.ts:820]: had to use a separate query here after the agencyId filters and before the page filtering.

File: express-api/src/services/projects/projectsServices.ts [hunk 2]
```diff
@@ -828,7 +830,8 @@ const getProjects = async (filter: ProjectFilter) => {
       logger.error('PropertyUnion Service - Invalid Sort Key');
     }
   }
-  return await query.getMany();
+  const projects = await query.getMany();
+  return { projects, totalCount };
```
Comment (dbarkowsky) [inline express-api/src/services/projects/projectsServices.ts:834]: There's an existing call that seems to work here and would do this all in one. 
```suggestion
  const [properties, totalCount] = await query.getManyAndCount();
  return { properties, totalCount };
```

File: express-api/src/services/projects/projectsServices.ts (hunk 1 above)
Comment (dbarkowsky) [reply express-api/src/services/projects/projectsServices.ts:820]: See the comment on the lines below. I think we could remove this if that works for you.

File: express-api/src/services/properties/propertiesServices.ts [hunk 3]
```diff
@@ -614,7 +616,8 @@ const getPropertiesUnion = async (filter: PropertyUnionFilter) => {
       logger.error('PropertyUnion Service - Invalid Sort Key');
     }
   }
-  return await query.getMany();
+  const properties = await query.getMany();
+  return { properties, totalCount };
```
Comment (dbarkowsky) [inline express-api/src/services/properties/propertiesServices.ts:620]: Same idea as with projects.
```suggestion
  const [properties, totalCount] = await query.getManyAndCount();
  return { properties, totalCount };
```

**Review by dbarkowsky — COMMENTED** [review]
I wonder if we need to snackbar popup on successful retrievals. The number of records is already shown above, so maybe that's duplicated info on screen. The error one could be useful though.

File: express-api/src/services/properties/propertiesServices.ts (hunk 3 above)
Comment (LawrenceLau2020) [reply express-api/src/services/properties/propertiesServices.ts]: Tested using getManyAndCount and it seems to work.

**Review by dbarkowsky — APPROVED** [review]
These changes are working well for me.
They don't apply to Agencies or Admin Areas yet, but those services may change a bit still in other tickets, so I'm not concerned. They currently show 0s.

**Pull request merged**